chainer/link.py

```diff
@@ -1205,18 +1195,7 @@ def __setitem__(self, index, value):  # NOQA
                 'ChainList indices must be integers or slices, not %s' %
                 type(index).__name__)
 
-    @tp.overload  # NOQA
-    def __getitem__(self, key):
-        # type: (int) -> Link
-        pass
-
-    @tp.overload  # NOQA
-    def __getitem__(self, key):
-        # type: (slice) -> collections_abc.MutableSequence[Link]
-        pass
-
-    def __getitem__(self, index):  # NOQA
-        # type: (tp.Union[int, slice]) -> tp.Union[Link, collections_abc.MutableSequence[Link]] # NOQA
```

Comment:
Should this hint be removed too (while it's kept for `__setitem__`)?

Reply:
Was explained offline that this removal is required (or else we need to use the overload decorator).